src/utils/geometry/delta-apply.ts

```diff
@@ -217,9 +217,19 @@ export function removeBoxTransforms(
  *
  * This is the final nested loop within HTMLVisualElement.updateLayoutDelta
  */
-export function applyTreeDeltas(box: AxisBox2D, treePath: HTMLVisualElement[]) {
+export function applyTreeDeltas(
+    box: AxisBox2D,
+    treeScale: Point2D,
+    treePath: HTMLVisualElement[]
+) {
     const treeLength = treePath.length
+    if (!treeLength) return
+
+    treeScale.x = treeScale.y = 1
     for (let i = 0; i < treeLength; i++) {
-        applyBoxDelta(box, treePath[i].delta)
+        const { delta } = treePath[i]
+        treeScale.x *= delta.x.scale
+        treeScale.y *= delta.y.scale
+        applyBoxDelta(box, delta)
```

Comment:
Since we're mutating treeScale and not doing anything with it, I'm not sure I understood what was going on here until I read the tests, I think because it's a separate file, so maybe it could benefit from a comment?